### ISSUE TITLE
Missing hour during DST long day when shifting or calculating ranges

### ISSUE THREAD

**Issue by jblackmersolea** [arrow-py/arrow#1162]
## Issue Description

When iterating or shifting over times that cross the DST long day change, an hour is skipped when the time zone is set for something other than UTC.

The following demonstrates the issue where the first loop prints 2 hours for the span, but converting those same timestamps to UTC prints 3 hours (correctly).  It's also strange that the tzinfo has to be set for the ISO timestamp containing the offset hours.

```
dts = arrow.get("2021-11-07T01:00:00-05:00", tzinfo="US/Central")
dte = arrow.get("2021-11-07T02:00:00-06:00", tzinfo="US/Central")
for dt in Arrow.range("hours", dts, dte):
    print(dt)
print("---BREAK---")
dts = arrow.get("2021-11-07T01:00:00-05:00", tzinfo="US/Central").to("UTC")
dte = arrow.get("2021-11-07T02:00:00-06:00", tzinfo="US/Central").to("UTC")
for dt in Arrow.range("hours", dts, dte):
    print(dt)
```

## System Info

- 🖥  Ubuntu 22.04.3 LTS
- 🐍  Python 3.10.12
- 🏹  Arrow 1.2.3

**Comment by ArcXeon** [comment]
Can I take this issue?

**Comment by SarthakNikhal** [comment]
Can I work on this issue? @jblackmersolea

**Comment by jblackmersolea** [comment]
Anyone can work on this issue as far as I know.

**Comment by Abdu-Ahmed** [comment]
Hey there!
I've opened a PR that fixes this: #1282

The root cause was that arrow.get() with an offset-aware string and a named tzinfo kwarg was replacing the timezone rather than converting,  causing the UTC moment to shift and fold information to be lost, the fix  converts using .to() instead of replacing, and also improves fromdatetime() to infer the correct fold when replacing tzinfo.
